libs/auth/scripts/check_pin.ts

```diff
@@ -1,58 +1,63 @@
 import { createInterface } from 'readline';
+import { extractErrorMessage, throwIllegalValue } from '@votingworks/basics';
+
 import { JavaCard } from '../src';
-import { CommonAccessCardDetails, CommonAccessCard } from '../src/cac';
+import { CommonAccessCard, CommonAccessCardDetails } from '../src/cac';
 import { CardDetails, PinProtectedCard, StatefulCard } from '../src/card';
 import { waitForReadyCardStatus } from './utils';
 
-/**
- * Checks whether a PIN is correct.
- */
-export async function main(args: readonly string[]): Promise<void> {
-  let card: PinProtectedCard &
-    StatefulCard<CardDetails | CommonAccessCardDetails>;
+const usageMessage = 'Usage: check-pin [--cac|--vxsuite (default)]';
 
-  for (const arg of args) {
-    switch (arg) {
-      case '--vxsuite': {
-        // default
-        break;
-      }
-
-      case '--cac': {
-        card = new CommonAccessCard();
-        break;
-      }
-
-      case '--help':
-      case '-h': {
-        process.stdout.write(`Usage: check-pin [--vxsuite (default)|--cac]\n`);
-        process.exit(0);
-        break;
-      }
-
-      default: {
-        process.stderr.write(`Unknown argument: ${arg}\n`);
-        process.exit(1);
-      }
-    }
+type CardType = 'cac' | 'vxsuite';
+
+function parseCommandLineArgs(args: readonly string[]): { cardType: CardType } {
+  if (![undefined, '--cac', '--vxsuite'].includes(args[0]) || args.length > 1) {
+    console.log(usageMessage);
+    process.exit(0);
   }
+  return { cardType: args[0] === '--cac' ? 'cac' : 'vxsuite' };
+}
 
+async function checkPin(cardType: CardType): Promise<void> {
   const pin = await new Promise<string>((resolve) => {
     createInterface(process.stdin, process.stdout).question(
       'Enter PIN: ',
       resolve
     );
   });
 
-  card ??= new JavaCard();
-  console.time('waitForReadyCardStatus');
-  await waitForReadyCardStatus(card);
-  console.timeEnd('waitForReadyCardStatus');
+  let card: PinProtectedCard &
+    StatefulCard<CardDetails | CommonAccessCardDetails>;
+  switch (cardType) {
+    case 'cac': {
+      card = new CommonAccessCard();
+      break;
+    }
+    case 'vxsuite': {
+      (process.env.VX_MACHINE_TYPE as string) = 'admin';
```

Comment:
The main update in this commit - previously, this had to be manually set. Rest of the updates are just for consistency with how other `libs/auth` scripts are structured